time-it-builder/src/lib.rs

```diff
@@ -37,37 +77,123 @@ impl TimeItBuilder {
     /// This function is called when the timer is done.
     /// The first argument is the name of the timer, the second is the duration.
     pub fn time_it(&self, action: TimeItCallBack) -> &Self {
-        let mut config = TIMER_CONFIG.write();
+        let mut config = TIMEIT_CONFIG.write();
         config.action = action;
         self
     }
 
     pub fn default_tag(&self, tag: impl ToString) -> &Self {
-        let mut config = TIMER_CONFIG.write();
+        let mut config = TIMEIT_CONFIG.write();
         config.default_tag = tag.to_string();
         self
     }
+
+    pub fn mode(&self, mode: TimeItMode) -> &Self {
+        let mut config = TIMEIT_CONFIG.write();
+        config.mode = mode;
+        self
+    }
 }
 
 pub fn get_config() -> &'static TimeItConfig {
-    LAZY_TIMER_CONFIG.get_or_create(|| TIMER_CONFIG.read().clone())
+    LAZY_TIMEIT_CONFIG.get_or_create(|| {
+        let mut config = TIMEIT_CONFIG.write();
+        let mut lazy_config = std::mem::replace(&mut *config, TimeItConfig::default());
```

Comment:
I'm taking it that this is supposed to be more performant than clone since the default is lightweight?

Reply:
I think this is just to get the mut ref out of the rwlock to put it in the lazyinit

Reply:
Also there's no need to clone here